nixos/modules/services/networking/bee/bee.nix

```diff
@@ -0,0 +1,272 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.bee;
+  format = pkgs.formats.yaml {};
+
+  deriveDataDir = id:
+    if id == null then "${cfg.dataDir}" else "${cfg.dataDir}/${toString id}";
+
+  makeSystemdTmpfilesEntry = id:
+    let
+      dataDir = deriveDataDir id;
+    in
+      [
+        "d '${dataDir}/'         0750 ${cfg.user} ${cfg.group}"
+      ];
+
+  makeSystemdServiceEntry = id:
+    let
+      dataDir        = deriveDataDir id;
+      namePostfix    = if id == null then "" else "-${toString id}";
+      serviceName    = "bee${namePostfix}";
+      configFile     = format.generate "bee.yaml" settings;
+      settings       = cfg.settings // {
```

Comment:
@SuperSandro2000 i could use some help/insight here: i'm generating a version of the configFile for each bee instance. but i also need those configFiles in the `bee-get-addr` script, but i don't know how to get it there.

maybe i should generate a copy of the `bee-get-addr` script for each instance? but i don't see how to merge their generation into this function, and/or how to get the configFile paths to where they would be generated.

any hints/ideas?

Reply:
You could either generate the script from here or call it with the right arguments.